src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
```

Comment:
`ZVAL_DEREF` and `SEPARATE_ZVAL_NOREF` are only done for "readPreferenceTags" because `php_phongo_read_preference_prep_tagsets()` may end up modifying the zval and we don't want those changes to affect the argument passed into `Manager::__construct()`. In this case, you're only reading the option zval's type code with `Z_TYPE_P()`, so you can skip any dereferencing and separation.